R/get_test_data.R

```diff
@@ -42,14 +42,16 @@ get_test_data <- function(recipe, x){
     stop("insufficient training data")
   }
 
+  groups <- epi_keys(recipe)[epi_keys(recipe) != "time_value"]
+
   test_data <- x %>%
     dplyr::filter(
       dplyr::if_any(
         .cols = recipe$term_info$variable[which(recipe$var_info$role == 'raw')],
         .fns = ~ !is.na(.x)
       )
     ) %>%
-    epiprocess::group_by(geo_value) %>%
+    epiprocess::group_by(across(groups)) %>%
```

Comment:
There's a tiny chance that the user will have defined a `groups` column, which will mess this up:
```
edf_with_groups_col = epiprocess::as_epi_df(tibble::tibble(geo_value="tx", groups=c("g1","g2","g3","g3","g3"), time_value=as.Date("2000-01-01")+0:4, value=1:5))
> edf_with_categories_col = epiprocess::as_epi_df(tibble::tibble(geo_value="tx", categories=c("g1","g2","g3","g3","g3"), time_value=as.Date("2000-01-01")+0:4, value=1:5))
> get_test_data(epi_recipe(edf_with_groups_col) %>% step_epi_ahead(value, ahead=0L) %>% step_epi_lag(value, lag=1L),
                edf_with_groups_col)
+ An `epi_df` object, with metadata:
* geo_type  = state
* time_type = day
* as_of     = 2022-06-15 06:18:59

# A tibble: 4 × 4
  geo_value time_value groups value
* <chr>     <date>     <chr>  <int>
1 tx        2000-01-01 g1         1
2 tx        2000-01-02 g2         2
3 tx        2000-01-04 g3         4
4 tx        2000-01-05 g3         5
> get_test_data(epi_recipe(edf_with_categories_col) %>% step_epi_ahead(value, ahead=0L) %>% step_epi_lag(value, lag=1L),
                edf_with_categories_col)
+ An `epi_df` object, with metadata:
* geo_type  = state
* time_type = day
* as_of     = 2022-06-15 06:19:25

# A tibble: 2 × 4
  geo_value time_value categories value
* <chr>     <date>     <chr>      <int>
1 tx        2000-01-04 g3             4
2 tx        2000-01-05 g3             5
```

It looks like using `across(tidyselect::all_of(groups))` would prevent this.

Reply:
Good catch and thank you for illustrating it with the example! I've updated the line now.